src/components/block-button.js

```diff
@@ -0,0 +1,31 @@
+window.blockUser = function(id) {
+  NAF.connection.adapter.publisher.handle.sendMessage({ kind: "block", whom: id });
+  NAF.connection.entities.removeEntitiesOfClient(id);
+};
+
+window.unblockUser = function(id) {
+  NAF.connection.adapter.publisher.handle.sendMessage({ kind: "unblock", whom: id });
+  window.setTimeout(() => {
+    // Wait enough time for the unblock to be processed
+    NAF.connection.entities.completeSync(id);
+  }, 1000);
```

Comment:
Hmm this seems odd. Shouldn't janus send a response back for this? (I assume sendMesage would resolve with a promise). Also same as above shouldnt this be handled by the event listeners you have set up on `document.body`?

Reply:
Oh, yes - `sendMessage` returns a promise. Will use that instead of this timeout.
No it shouldn't be handled by the event listener because when client A unblocks client B, only client B receives the "unblocked" message that is being handled by the event listeners.

Reply:
Oh, I misunderstood your earlier suggestion. Yes, the event listeners can handle it for both.